blackjax/util.py

```diff
@@ -142,12 +143,15 @@ def index_pytree(input_pytree: ArrayLikeTree) -> ArrayTree:
 
 def run_inference_algorithm(
     rng_key: PRNGKey,
-    initial_state_or_position: ArrayLikeTree,
     inference_algorithm: Union[SamplingAlgorithm, VIAlgorithm],
     num_steps: int,
+    initial_state: ArrayLikeTree = None,
+    initial_position: ArrayLikeTree = None,
     progress_bar: bool = False,
     transform: Callable = lambda x: x,
-) -> tuple[State, State, Info]:
+    return_state_history=True,
+    expectation: Callable = lambda x: x,
```

Comment:
Should we combine the kwargs of `transform` and `expectation` for now? Otherwise we need a better name for `expectation`

Reply:
`expectation` seems to me like an appropriate name, since indeed the value calculated is the expectation. For example, if `expectation=lambda x: x**2`, then you get back $E[x^2]$.

By contrast, `transform` operates on the full history of samples. In the future, I think it would make sense for it to also take Info, so that the user can choose to dispense with (part of) the diagnostic info. That is why I thought it was better to keep them separate.

I think it would in theory be possible to make them the same, if necessary.

Reply:
That makes sense, but looking at the implementation, the return state is always `transform(state)`, which means for a expectation transformation, we should probably do `expectation(transform(x))`?